File: wiki-spark/java/src/main/java/utils/ISO8601.java
package utils;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.TimeZone;

/**
 * Helper class for handling ISO 8601 strings of the following format:
 * "2008-03-01T13:00:00+01:00". It also supports parsing the "Z" timezone.
 */
public final class ISO8601 {
	/** Transform timestamp(ms) to Calendar */
	public static String timestampToString(final long timestamp) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTimeInMillis(timestamp);
		Date date = calendar.getTime();

		SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
		sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
		String formatted = sdf.format(date);

		return formatted;
	}

	/** Transform ISO 8601 string to timestamp. */
	public static long toTimeMS(final String iso8601string)
			throws ParseException {
		String s = iso8601string.replace("Z", "+00:00");
		try {
			s = s.substring(0, 22) + s.substring(23); // to get rid of the ":"
		} catch (IndexOutOfBoundsException e) {
			throw new ParseException("Invalid length", 0);
		}
		Date date = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").parse(s);

		return date.getTime();
	}
}

File: wiki-spark/scala/src/main/scala/utils/ISO8601.scala
package utils

import java.text.ParseException

import java.text.SimpleDateFormat

import java.util.Calendar

import java.util.Date

import java.util.TimeZone

//remove if not needed
import scala.collection.JavaConversions._

object ISO8601 {

  /**
   * Transform timestamp(ms) to Calendar
   */
  def timestampToString(timestamp: Long): String = {
    val calendar: Calendar = Calendar.getInstance
    calendar.setTimeInMillis(timestamp)
    val date: Date = calendar.getTime
    val sdf: SimpleDateFormat = new SimpleDateFormat(
      "yyyy-MM-dd'T'HH:mm:ss'Z'")
    sdf.setTimeZone(TimeZone.getTimeZone("GMT"))
    val formatted: String = sdf.format(date)
    formatted
  }

  /**
   * Transform ISO 8601 string to timestamp.
   */
  def toTimeMS(iso8601string: String): Long = {
    var s: String = iso8601string.replace("Z", "+00:00")
    s = s.substring(0, 22) + s.substring(23)
    val date: Date = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").parse(s)
    date.getTime
  }

}
